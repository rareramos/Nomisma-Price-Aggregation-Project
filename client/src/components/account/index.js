export { Account as default} from './account-container';
