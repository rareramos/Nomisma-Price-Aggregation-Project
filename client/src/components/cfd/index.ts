export { Cfd } from './cfd-container';
