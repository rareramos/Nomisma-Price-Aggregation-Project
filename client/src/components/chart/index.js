export { Chart } from './chart-container';
