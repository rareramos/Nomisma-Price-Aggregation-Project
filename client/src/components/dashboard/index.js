export { Dashboard } from './dashboard-container';
