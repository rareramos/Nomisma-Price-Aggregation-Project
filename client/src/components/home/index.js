export { Home as default } from './home-container';
