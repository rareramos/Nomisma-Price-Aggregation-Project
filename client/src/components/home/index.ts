export { Home } from './home-container';
