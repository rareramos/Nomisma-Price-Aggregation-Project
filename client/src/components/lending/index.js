export { Lending } from './lending-container';
