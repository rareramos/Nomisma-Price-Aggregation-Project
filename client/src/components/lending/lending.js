import React, { Component, Fragment } from 'react';
import {
  bool, arrayOf, string,
  number, func, object, shape, objectOf,
} from 'prop-types';
import { IconSortUp, IconSortDown } from '@nomisma/nomisma-ui/icons';
import { Table } from 'reactstrap';
import Pagination from 'react-js-pagination';

import { Loading } from '../common/loading/index';
import { Styled } from './styled';
import { TableToolbar } from '../table-toolbar';
import { Chart } from '../chart';

export class Lending extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activePage: 1,
    };
    this.PageChange = this.PageChange.bind(this);
  }

  PageChange(pagenumber) {
    const { handlePageChange } = this.props;

    this.setState({ activePage: pagenumber });
    handlePageChange(pagenumber);
  }

  static renderSortIcon(order) {
    const style = { marginLeft: '0.25rem', fill: 'var(--primary)' };
    if (order === 1) {
      return <IconSortUp style={style} />;
    } if (order === -1) {
      return <IconSortDown style={style} />;
    }
    return null;
  }

  render() {
    const {
      loading,
      tableData,
      pageCount,
      columns,
      sortByColumn,
      filter,
    } = this.props;
    const { activePage } = this.state;
    const itemCounts = (pageCount - 1) * 15;
    return (
      <Styled>
        <Chart />
        <TableToolbar />
        {
          /* eslint-disable-next-line no-nested-ternary */
          loading ? <Loading fill="var(--primary)" /> : tableData.length > 0 ? (
            <Fragment>
              <Table>
                <thead>
                  <tr>
                    { columns.map(({ key, columnName }) => (
                      <th
                        key={columnName}
                        onClick={() => sortByColumn(key)}
                      >
                        { columnName }
                        { filter.sort === key ? Lending.renderSortIcon(filter.order) : null }
                      </th>
                    )) }
                  </tr>
                </thead>
                <tbody>
                  { tableData.map(
                    (
                      item,
                      idx,
                    ) => (
                      // eslint-disable-next-line react/no-array-index-key
                      <tr key={idx}>
                        { columns.map(
                          (
                            {
                              key,
                            },
                          ) => (
                            // eslint-disable-next-line react/no-array-index-key
                            <td key={`${key}-${idx}`}>
                              { item[key] }
                            </td>
                          ),
                        )
                        }
                      </tr>
                    ),
                  )
                  }
                </tbody>
              </Table>
              <Pagination
                activePage={activePage}
                itemsCountPerPage={15}
                totalItemsCount={itemCounts}
                pageRangeDisplayed={5}
                onChange={this.PageChange}
              />
            </Fragment>
          ) : <p>No results found...</p>
        }
      </Styled>
    );
  }
}

Lending.defaultProps = {
  tableData: [],
};

Lending.propTypes = {
  loading: bool.isRequired,
  pageCount: number.isRequired,
  tableData: arrayOf(object),
  handlePageChange: func.isRequired,
  columns: arrayOf(shape({
    columnName: string, key: string, render: func, selector: func,
  })).isRequired,
  sortByColumn: func.isRequired,
  filter: objectOf(shape({ sort: string, order: number })).isRequired,
};
