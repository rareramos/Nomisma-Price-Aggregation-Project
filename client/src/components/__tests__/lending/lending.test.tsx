import React from 'react';
import { shallow } from 'enzyme';

import { Lending as LendingComp } from 'components/lending/lending';
import { ILendingProps } from 'types/components/lending';


const LendingProps : ILendingProps = {
  columns: [],
  filter: {},
  handlePageChange: jest.fn,
  loading: true,
  pageCount: 25,
  sortByColumn: jest.fn,
  tableData: [{}],
};

describe('Lending test', () => {
  it('matches lending component snapshot', () => {
    const wrapper = shallow(<LendingComp {...LendingProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
