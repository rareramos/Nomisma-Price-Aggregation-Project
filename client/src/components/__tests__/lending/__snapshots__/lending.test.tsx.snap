// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lending test matches lending component snapshot 1`] = `
<styled.div>
  <Connect(ChartView) />
  <TableToolbar />
  <Loading
    fill="var(--primary)"
  />
</styled.div>
`;
